File: server/internal/business/models/account.go
package models

import (
	"database/sql/driver"
	"fmt"
	"time"

	"github.com/google/uuid"
)

type AccountType string

const (
	AccountTypeBank     AccountType = "bank"
	AccountTypeCash     AccountType = "cash"
	AccountTypeAlipay   AccountType = "alipay"
	AccountTypeWeChat   AccountType = "wechat"
	AccountTypeCredit   AccountType = "credit"
	AccountTypeInvestment AccountType = "investment"
	AccountTypeOther    AccountType = "other"
)

type Account struct {
	ID             uuid.UUID   `db:"id" json:"id"`
	UserID         uuid.UUID   `db:"user_id" json:"user_id"`
	Name           string      `db:"name" json:"name"`
	Type           AccountType `db:"type" json:"type"`
	Currency       string      `db:"currency" json:"currency"`
	Balance        float64     `db:"balance" json:"balance"`
	CreatedAt      time.Time   `db:"created_at" json:"created_at"`
	UpdatedAt      time.Time   `db:"updated_at" json:"updated_at"`
	LastModifiedAt time.Time   `db:"last_modified_at" json:"last_modified_at"`
	Version        int         `db:"version" json:"version"`
	IsDeleted      bool        `db:"is_deleted" json:"is_deleted"`
}

func (a *AccountType) Scan(value interface{}) error {
	str, ok := value.(string)
	if !ok {
		return fmt.Errorf("invalid account type: %v", value)
	}
	*a = AccountType(str)
	return nil
}

func (a AccountType) Value() (driver.Value, error) {
	return string(a), nil
}

File: flutter/lib/data/models/account.dart
import 'package:equatable/equatable.dart';
import 'package:json_annotation/json_annotation.dart';

import '../../sync/lww_strategy.dart';

part 'account.g.dart';

enum AccountType {
  @JsonValue('bank')
  bank,
  @JsonValue('cash')
  cash,
  @JsonValue('alipay')
  alipay,
  @JsonValue('wechat')
  wechat,
  @JsonValue('credit')
  credit,
  @JsonValue('investment')
  investment,
  @JsonValue('other')
  other,
}

@JsonSerializable()
class Account extends Equatable implements LwwEntity {
  final String id;
  @JsonKey(name: 'user_id')
  final String userId;
  final String name;
  final AccountType type;
  final String currency;
  final double balance;
  @JsonKey(name: 'created_at')
  final DateTime createdAt;
  @JsonKey(name: 'updated_at')
  final DateTime updatedAt;
  @JsonKey(name: 'last_modified_at')
  final DateTime lastModifiedAt;
  final int version;
  @JsonKey(name: 'is_deleted')
  final bool isDeleted;

  const Account({
    required this.id,
    required this.userId,
    required this.name,
    required this.type,
    required this.currency,
    required this.balance,
    required this.createdAt,
    required this.updatedAt,
    required this.lastModifiedAt,
    required this.version,
    required this.isDeleted,
  });

  Account copyWith({
    String? id,
    String? userId,
    String? name,
    AccountType? type,
    String? currency,
    double? balance,
    DateTime? createdAt,
    DateTime? updatedAt,
    DateTime? lastModifiedAt,
    int? version,
    bool? isDeleted,
  }) {
    return Account(
      id: id ?? this.id,
      userId: userId ?? this.userId,
      name: name ?? this.name,
      type: type ?? this.type,
      currency: currency ?? this.currency,
      balance: balance ?? this.balance,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
      lastModifiedAt: lastModifiedAt ?? this.lastModifiedAt,
      version: version ?? this.version,
      isDeleted: isDeleted ?? this.isDeleted,
    );
  }

  factory Account.fromJson(Map<String, dynamic> json) => _$AccountFromJson(json);

  Map<String, dynamic> toJson() => _$AccountToJson(this);

  @override
  List<Object?> get props => [
    id, userId, name, type, currency, balance, createdAt,
    updatedAt, lastModifiedAt, version, isDeleted,
  ];
}
